### PR TITLE
Get player email through API

### PR DESCRIPTION
**Pull request by YellowZaki**
Relates to https://github.com/AuthMe/AuthMeReloaded/issues/1373 https://github.com/AuthMe/AuthMeReloaded/issues/1448

### CODE REVIEW

**Comment by YellowZaki** [comment]
Well, I didn't realised default email is your@email.com, not null.

**Comment by ljacqu** [comment]
Sorry for the late reply.
I think it would make sense to null out the email as well in the above method if it's the default. Isn't there a constant on PlayerAuth that has the value `your@email.com`?

In the future I'd like to go in the direction of having some sort of interface we'll publicly expose with the information (name / email / etc.) so that a user will just get some sort of ApiUser object and doesn't have to call multiple methods that load the PlayerAuth each time to get basic info. But I'm happy to have these methods for now since I don't think that will happen soon... :P But with that in mind, maybe it doesn't make sense to have the `hasEmail` method.

**Comment by YellowZaki** [comment]
There is no constant for your@email.com.
I have made that getEmail() returns null if default.
I think it is good to keep hasEmail(), it is more user friendly.

**Comment by ljacqu** [comment]
PlayerAuth#DB_EMAIL_DEFAULT 

The thing about the other method is, yes, it's a nice readable method, but it doesn't add new functionality. I'd be on board with such a method if we had a dedicated class to users or something, but since we anyway intend on providing an alternative to the getEmail method, I'd like to keep the number of methods to a minimum that a user has to be familiar with. Again, only because we intend to provide another approach later on and because we only have one API class.

File: src/main/java/fr/xephi/authme/api/v3/AuthMeApi.java [hunk 1]
```diff
@@ -352,4 +352,36 @@ public String getCountryCode(String ip) {
...
+     * @return The email of the player
+     */
+    public String getEmail(String playerName) {
+        PlayerAuth auth = playerCache.getAuth(playerName);
+        if (auth == null) {
+            auth = dataSource.getAuth(playerName);
+        }
+        if (auth != null) {
+            String email = auth.getEmail();
+            if (("your@email.com").equals(email)) {
```
Comment (sgdc3) [inline src/main/java/fr/xephi/authme/api/v3/AuthMeApi.java]: PlayerAuth#DB_EMAIL_DEFAULT

File: src/main/java/fr/xephi/authme/api/v3/AuthMeApi.java [hunk 2]
```diff
@@ -352,4 +352,36 @@ public String getCountryCode(String ip) {
...
+    }    
+    
+    /**
+     * Check if player has linked email
+     *
+     * @param playerName The name of the player to check
+     * @return true if the player has linked email
+     */
+    public boolean hasEmail(String playerName) {
+        return (getEmail(playerName) != null);
```
Comment (sgdc3) [inline src/main/java/fr/xephi/authme/api/v3/AuthMeApi.java]: Don't wrap it with ()

**Comment by YellowZaki** [comment]
Since https://github.com/AuthMe/AuthMeReloaded/pull/2000 has been implemented, I close this

**Comment by ljacqu** [comment]
Sorry for the lack of update, and thanks for your contribution :) This is what sparked the addition of the player API :)

**Pull request closed without merging**